.github/workflows/fedora_29.yml

```diff
@@ -0,0 +1,231 @@
+
+      - name: Show affected components
+        run: |
+          echo "${{ toJSON(steps.smart_ci.outputs.affected_components) }}"
+        shell: bash
+
+  Docker:
+    needs: Smart_CI
+    runs-on: aks-linux-4-cores-16gb-docker-build
+    container:
+      image: openvinogithubactions.azurecr.io/docker_build:0.2
+      volumes:
+        - /mount:/mount
+    outputs:
+      images: "${{ steps.handle_docker.outputs.images }}"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - uses: ./.github/actions/handle_docker
+        id: handle_docker
+        with:
+          images: |
+            ov_build/fedora_29
+          registry: 'openvinogithubactions.azurecr.io'
+          dockerfiles_root_dir: '.github/dockerfiles'
+          changed_components: ${{ needs.smart_ci.outputs.changed_components }}
+
+  Build:
+    needs: [Docker, Smart_CI]
+    if: "!needs.smart_ci.outputs.skip_workflow"
+    uses: ./.github/workflows/job_build_linux.yml
+    with:
+      runner: 'aks-linux-16-cores-32gb'
+      container: '{"image": "${{ fromJSON(needs.docker.outputs.images).ov_build.fedora_29 }}", "volumes": ["/mount:/mount"], "options": "-e SCCACHE_AZURE_BLOB_CONTAINER -e SCCACHE_AZURE_CONNECTION_STRING"}'
+      affected-components: ${{ needs.smart_ci.outputs.affected_components }}
+      event-name: ${{ github.event_name }}
+      os: 'fedora_29'
+      build-contrib: false
+      build-debian-packages: false
+      build-js: false
+      build-rpm-packages: false
+      cmake-options: |- 
+            -G "Ninja" \
+            -DENABLE_CPPLINT=OFF \
+            -DENABLE_NCC_STYLE=OFF \
+            -DENABLE_TESTS=ON \
+            -DENABLE_STRICT_DEPENDENCIES=OFF \
+            -DENABLE_SYSTEM_OPENCL=ON \
+            -DENABLE_PYTHON_PACKAGING=ON \
+            -DCPACK_GENERATOR=TGZ \
+            -DENABLE_INTEL_GPU=OFF \
+            -DENABLE_INTEL_NPU=OFF \
+            -DCMAKE_COMPILE_WARNING_AS_ERROR=OFF \
+            -DCMAKE_BUILD_TYPE=Release
+
+  CXX_Unit_Tests:
+    name: C++ unit tests
+    needs: [Smart_CI, Docker, Build]
+    timeout-minutes: 30
+    runs-on: 'aks-linux-16-cores-32gb'
+    container:
+      image: ${{ fromJSON(needs.docker.outputs.images).ov_build.fedora_29 }}
+    defaults:
+      run:
+        shell: 'bash'
+    env:
+      INSTALL_DIR: ${{ github.workspace }}/install
+      INSTALL_TEST_DIR: ${{ github.workspace }}/install/tests
+      SOURCE_COMMAND: 'source'
+      SETUPVARS: 'setupvars.sh'
+    steps:
+      - name: Download OpenVINO package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: openvino_package
+          path: ${{ env.INSTALL_DIR }}
+
+      - name: Download OpenVINO tests package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: openvino_tests
+          path: ${{ env.INSTALL_TEST_DIR }}
```

Comment:
why only subset of tests is used?

Reply:
This is a subset that is executed in Jenkins. I tried executing the full scope but encountered problems on the steps that are outside this scope, e.g.: https://github.com/openvinotoolkit/openvino/actions/runs/10720137452/job/29727562252. Do you think the full scope should be executed and some stages skipped?

Reply:
Some dependencies were not installed https://github.com/openvinotoolkit/openvino/actions/runs/10720137452/job/29725836109#step:10:351

Please, update pip here https://github.com/openvinotoolkit/openvino/blob/master/.github/workflows/job_build_linux.yml#L135 using:
```
pip3 install -U pip
```

Reply:
Thanks, I've included it in the build job and will see how it goes.

Though, should not installing pip in Dockerfile (https://github.com/openvinotoolkit/openvino/pull/26336/files#diff-f9cc5581e25b3490c8e26bef7dbfc3bd7329ffc1adbbbc4b6226ebde354bdd0aR48) be enough?

Reply:
> Though, should not installing pip in Dockerfile (https://github.com/openvinotoolkit/openvino/pull/26336/files#diff-f9cc5581e25b3490c8e26bef7dbfc3bd7329ffc1adbbbc4b6226ebde354bdd0aR48) be enough?

good question. Does creation of venv inherit pip version?

Reply:
Strangely enough, pip's version is 20, I'll take a look in the Dockerfile and maybe we could avoid doing the upgrading in the build job.

Reply:
Yeah, the problem was with the venv being created before the pip upgrade. I've changed the order and all seems to be working now, including the full C++ test scope.